File: app/src/main/java/net/north101/android/ghplayertracker/CharacterViewHolder.java
package net.north101.android.ghplayertracker;

import android.graphics.Color;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import net.north101.android.ghplayertracker.data.SelectableCharacter;

public class CharacterViewHolder extends BaseViewHolder<SelectableCharacter> {
    public static int layout = R.layout.character_item_view;

    private final ImageView iconView;
    private final TextView nameView;
    private final TextView levelView;

    public static CharacterViewHolder inflate(ViewGroup parent) {
        return new CharacterViewHolder(inflate(parent, layout));
    }

    public CharacterViewHolder(View view) {
        super(view);

        this.iconView = view.findViewById(R.id.class_icon);
        this.nameView = view.findViewById(R.id.name);
        this.levelView = view.findViewById(R.id.level);
    }

    public void bind(SelectableCharacter item) {
        super.bind(item);

        this.iconView.setImageResource(Util.getImageResource(itemView.getContext(), item.character.getCharacterClass().id));
        this.nameView.setText(item.character.getName());
        this.levelView.setText(String.valueOf(item.character.getCurrentLevel().level));
        if (item.selected) {
            itemView.setBackgroundColor(Color.LTGRAY);
        } else {
            itemView.setBackgroundColor(Color.WHITE);
        }
    }
}

File: app/src/main/java/net/north101/android/ghplayertracker/CharacterViewHolder.kt
package net.north101.android.ghplayertracker

import android.graphics.Color
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView

import net.north101.android.ghplayertracker.data.SelectableCharacter

class CharacterViewHolder(view: View) : BaseViewHolder<SelectableCharacter>(view) {
    private val iconView: ImageView = view.findViewById(R.id.icon)
    private val nameView: TextView = view.findViewById(R.id.name)
    private val levelView: TextView = view.findViewById(R.id.level)

    override fun bind(item: SelectableCharacter) {
        super.bind(item)

        this.iconView.setImageResource(Util.getImageResource(itemView.context, item.character.characterClass.id))
        this.nameView.text = item.character.name
        this.levelView.text = item.character.currentLevel.level.toString()
        if (item.selected) {
            itemView.setBackgroundColor(Color.LTGRAY)
        } else {
            itemView.setBackgroundColor(Color.WHITE)
        }
    }

    companion object {
        var layout = R.layout.character_item_view

        fun inflate(parent: ViewGroup): CharacterViewHolder {
            return CharacterViewHolder(BaseViewHolder.inflate(parent, layout))
        }
    }
}
